libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+
+        # Authentication state,
+        # one of "guest", "untrusted" or "trusted"
+        self.auth = None
+
+        self.config = None
+
+        # Various information about the host (OS, kernel, ...)
+        self.environment = None
+
+        self.public = None
+
+    def __str__(self):
+        return str(self.api_extensions) + str(self.api_status) + \
+            str(self.api_version) + str(self.auth) + str(self.config) + \
+            str(self.environment) + \
+            str(self.public)
+
+
+class LXDResponse(JsonResponse):
+    valid_response_codes = [httplib.OK, httplib.ACCEPTED, httplib.CREATED,
+                            httplib.NO_CONTENT]
+
+    def parse_body(self):
+
+        if len(self.body) == 0 and not self.parse_zero_length_body:
+            return self.body
+
+        try:
+            # error responses are tricky in Docker. Eg response could be
+            # an error, but response status could still be 200
+            content_type = self.headers.get('content-type', 'application/json')
+            if content_type == 'application/json' or content_type == '':
+                if self.headers.get('transfer-encoding') == 'chunked' and \
+                        'fromImage' in self.request.url:
+                    body = [json.loads(chunk) for chunk in
+                            self.body.strip().replace('\r', '').split('\n')]
+                else:
+                    body = json.loads(self.body)
+            else:
+                body = self.body
+        except ValueError:
+            m = re.search('Error: (.+?)"', self.body)
+            if m:
+                error_msg = m.group(1)
+                raise Exception(error_msg)
+            else:
+                raise Exception(
+                    'ConnectionError: Failed to parse JSON response')
+        return body
+
+    def parse_error(self):
+        if self.status == 401:
+            raise InvalidCredsError('Invalid credentials')
+        return self.body
+
+    def success(self):
+        return self.status in self.valid_response_codes
+
+
+class LXDConnection(ConnectionUserAndKey):
+    responseCls = LXDResponse
+    timeout = 60
+
+    def add_default_headers(self, headers):
+        """
+        Add parameters that are necessary for every request
+        If user and password are specified, include a base http auth
+        header
+        """
+        headers['Content-Type'] = 'application/json'
+        if self.user_id and self.key:
+            user_b64 = base64.b64encode(b('%s:%s' % (self.user_id, self.key)))
+            headers['Authorization'] = 'Basic %s' % (user_b64.decode('utf-8'))
+        return headers
+
+
+class LXDtlsConnection(KeyCertificateConnection):
+
+    responseCls = LXDResponse
+
+    def __init__(self, key, secret, secure=True,
+                 host='localhost', port=8443, ca_cert='',
+                 key_file=None, cert_file=None, **kwargs):
+
+        if 'certificate_validator' in kwargs.keys():
```

Comment:
Please explicitly declare supported arguments (``certificate_validator``) instead of using ``**kwargs`` (I know a lot of the existing drivers did, but we just tried to clean and improve some of that legacy code).

Reply:
basically I am seriously thinking to remove this altogether...I put it there initially to get more informative errors when trying to use TLS. I am happy to so as you suggested but what do you think about removing it completely?